src/pke/include/cryptocontext.h

```diff
@@ -169,6 +179,17 @@ class CryptoContextImpl : public Serializable {
     Plaintext MakePlaintext(const PlaintextEncodings encoding, const std::vector<int64_t>& value, size_t depth,
                             uint32_t level) const {
         const auto cryptoParams = std::dynamic_pointer_cast<CryptoParametersRNS>(GetCryptoParameters());
+        // validation of level: We need to compare it to multiplicativeDepth, but multiplicativeDepth is not
+        // readily available. what we can get is numModuli, and numModuli is (multiplicativeDepth+1)
+        size_t numModuli             = cryptoParams->GetElementParams()->GetParams().size();
+        uint32_t multiplicativeDepth = numModuli - 1;
```

Comment:
This equation is not true for BFV. It is only true for BGV and CKKS. However, in all three cases we do need level >= numModuli.

Reply:
A correction. Level = 0 should be used for BFV. For BGV and CKKS, this logic can be used. However, if the scaling mode is set to FLEXIBLEAUTOEXT, the multiplicative depth = numModuli - 2.

Reply:
The comments have been addressed